core/primitives/src/bandwidth_scheduler.rs

```diff
@@ -82,6 +82,55 @@ pub struct BandwidthRequest {
 /// There are this many predefined values of bandwidth that can be requested in a BandwidthRequest.
 pub const BANDWIDTH_REQUEST_VALUES_NUM: usize = 40;
 
+/// Values of bandwidth that can be requested in a bandwidth request.
+/// When the nth bit is set in a request bitmap, it means that a shard is requesting the nth value from this list.
+/// The list is sorted, from smallest to largest values.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct BandwidthRequestValues {
+    pub values: [Bandwidth; BANDWIDTH_REQUEST_VALUES_NUM],
+}
+
+impl BandwidthRequestValues {
+    pub fn new(params: &BandwidthSchedulerParams) -> BandwidthRequestValues {
+        // values[-1] = base_bandwidth
+        // values[values.len() - 1] = max_shard_bandwidth
+        // values[i] = linear interpolation between values[-1] and values[values.len() - 1]
+        let mut values = [0; BANDWIDTH_REQUEST_VALUES_NUM];
+
+        for i in 0..values.len() {
+            let values_len_u64: u64 =
+                values.len().try_into().expect("Converting usize to u64 shouldn't fail");
+            let i_u64: u64 = i.try_into().expect("Converting usize to u64 shouldn't fail");
+
+            values[i] = params.base_bandwidth
+                + (params.max_shard_bandwidth - params.base_bandwidth) * (i_u64 + 1)
+                    / values_len_u64;
```

Comment:
nit: maybe make that a methon in the params? or just implement a `interpolate` method that is a bit more generic and call it from here
`interpolate(min, max, i, n)`

Reply:
> maybe make that a methon in the params

I wanted to keep params separate from values to allow many different formats of requests.
Params are pretty generic, describe how much a shard can send etc.
Values are specific to the request format. Maybe one day we'll move to exponential values and it'll be a separate values struct that uses the same params.

Reply:
> or just implement a interpolate method that is a bit more generic and call it from here

Added a generic function. Nice, it made things more readable :+1: